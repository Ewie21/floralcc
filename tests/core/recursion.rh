void put_array(int arr, int i, int len) {
    put(*(arr - 8 * i));
    if (i != len - 1) {
        //pot(i + 1);
        put_array(arr, i + 1, len);
    }
}

// arr: array
// i: iterator (start at 0 for whole array)
// n: increment number
// len: length of array
void increment_array(int arr, int i, int n, int len) {
    arr[i] += 1;
    put(arr[i]);
    if (i != len - 1) {
        i += 1;
        increment_array(arr, i, n, len);
    }
}

// infinite loop if start > end
void basic_recursion_test(int i, int end) {
    put(i);
    if (i != end) {
        i += 1;
        basic_recursion_test(i, end);
    }
}

void put_ptr(int ptr) {
    put(*ptr);
}

void nested_call_test(int ptr, int ptr_two) {
    put_ptr(ptr);
    put_ptr(ptr_two);
}

int t = '6';
int o = '7';
int ptr = &t;
int ptr_two = &o;

int h = [3; '0', '1', '2'];
increment_array(h, 0, 1, 3);
put('\n');
basic_recursion_test('6', '8');
put('\n');
nested_call_test(ptr, ptr_two);
put('\n');
put('l');
