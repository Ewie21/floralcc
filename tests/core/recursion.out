123
678
67
l
